### DISCUSSION TITLE
New Storybook 7 stories indexer refresh question

### DISCUSSION THREAD

**Discussion by christianwiedemann** [storybookjs/storybook#21039] [Help]
Hi!
I am currently developing an indexer which loads the stories from yaml file with the stories indexer in storybook 7 (beta43). Now I am looking for a way to refresh the stories if the yaml file changed. Is there something like addDependency?

Thanks!

**Comment by psychobolt** [comment]
Are you creating a plugin for your builder framework (Webpack, Vite) that transforms your YAML to CSF? In that case I expect when the JSON is changed and HMR is fired, the builder should trigger the plugin to reimport JSON. Don't forget about cache busting...

**Comment by valentinpalkovic** [comment]
@shilman  Do you have a suggestion or an idea how refresh would work in the above mentioned scenario?

**Reply by shilman** [reply]
The indexer watches the globs in your main.js `stories` field. If any matching files change, it will re-run and trigger the sidebar to update.

The currently displayed story updates based on HMR, which is provided by the builder (webpack or vite) and the indexer is not involved.
